demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,625 @@
+
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+/* Azure Provisioning/IoTHub library includes */
+#include "azure_iot_hub_client.h"
+#include "azure_iot_provisioning_client.h"
+
+/* Exponential backoff retry include. */
+#include "backoff_algorithm.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "transport_tls_socket.h"
+
+/* Crypto helper header. */
+#include "crypto.h"
+/*-----------------------------------------------------------*/
+
+/* Compile time error for undefined configs. */
+#if !defined( democonfigHOSTNAME ) && !defined( democonfigENABLE_DPS_SAMPLE )
+    #error "Define the config democonfigHOSTNAME by following the instructions in file demo_config.h."
```

Comment:
Is there a `demo_config.h` that will be going to come in a separate PR?  If so it's perfectly fine, but if not we should discuss offline how customer would get this.

Reply:
yes, demo_config.h is something that is application specific and will be part of separate PR.